onnx/version_converter/convert.h

```diff
@@ -105,120 +107,341 @@ class DefaultVersionConverter : public BaseVersionConverter {
+      registerAdapter(make_unique<TypeRestriction>("Tanh",
+        OpSetID(6), OpSetID(5), broadcast_unallowed_types));
+      registerAdapter(make_unique<BatchNormalization_6_5>());
+
+      /******** 6 -> 7 ********/
+      registerAdapter(make_unique<CompatibleAdapter>("AveragePool",
+        OpSetID(6), OpSetID(7)));      
+      registerAdapter(make_unique<BroadcastForwardCompatibility>("Add",
+        OpSetID(6), OpSetID(7)));
+      registerAdapter(make_unique<BroadcastForwardCompatibility>("Div",
+        OpSetID(6), OpSetID(7)));
+      registerAdapter(make_unique<BroadcastForwardCompatibility>("Mul",
+        OpSetID(6), OpSetID(7)));
+      registerAdapter(make_unique<BroadcastForwardCompatibility>("Pow",
+        OpSetID(6), OpSetID(7)));
+      registerAdapter(make_unique<CompatibleAdapter>("PRelu",
+        OpSetID(6), OpSetID(7)));    
+      registerAdapter(make_unique<BroadcastForwardCompatibility>("Sub",
+        OpSetID(6), OpSetID(7)));
+      registerAdapter(make_unique<Gemm_6_7>());
+      registerAdapter(make_unique<BatchNormalization_6_7>());
+      registerAdapter(make_unique<Dropout_6_7>());
+
+      /******** 7 -> 6 ********/   
+      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Add",
+        OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Div",
+        OpSetID(7), OpSetID(6)));
       registerAdapter(make_unique<BroadcastBackwardCompatibility>("Mul",
         OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Pow",
+        OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<CompatibleAdapter>("PRelu",
+        OpSetID(7), OpSetID(6)));  
+      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Sub",
+        OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<SetIsTest>("BatchNormalization",
+        OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<SetIsTest>("Dropout",
+        OpSetID(7), OpSetID(6)));
+      registerAdapter(make_unique<Gemm_7_6>());
+      registerAdapter(make_unique<AveragePool_7_6>());
+
+      /******** 7 -> 8 ********/  
       registerAdapter(make_unique<CompatibleAdapter>("Max",
         OpSetID(7), OpSetID(8)));
-      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Max",
-        OpSetID(8), OpSetID(7)));
       registerAdapter(make_unique<CompatibleAdapter>("Min",
         OpSetID(7), OpSetID(8)));
-      registerAdapter(make_unique<BroadcastBackwardCompatibility>("Min",
-        OpSetID(8), OpSetID(7)));
       registerAdapter(make_unique<CompatibleAdapter>("Mean",
         OpSetID(7), OpSetID(8)));
+      registerAdapter(make_unique<CompatibleAdapter>("Sum",
+        OpSetID(7), OpSetID(8)));
+      registerAdapter(make_unique<CompatibleAdapter>("MaxPool",
+        OpSetID(7), OpSetID(8)));
+
+      /******** 8 -> 7 ********/  
+      /* Should the adapters for Max, Min, Mean be CompatiblAdapter here? */
```

Comment:
The three adapters for `"Max"`, `"Min"` and `"Mean"` here were pre-existing as `BroadcastBackwardCompatibility`, but it really looks they should be `CompatibleAdapter`. What is the right adapter?

Reply:
According to https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Mean-8,
(same as Max-8 and Min-8), they started to support broadcasting from opset8. 
In that case, using `BroadcastBackwardCompatibility` here makes more sense to me.

Reply:
Indeed. Removing the comment then